src/librustc_mir_build/build/expr/as_temp.rs

```diff
@@ -59,10 +59,18 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             if let Some(tail_info) = this.block_context.currently_in_block_tail() {
                 local_decl = local_decl.block_tail(tail_info);
             }
-            if let ExprKind::StaticRef { def_id, .. } = expr.kind {
-                let is_thread_local = this.hir.tcx().has_attr(def_id, sym::thread_local);
-                local_decl.internal = true;
-                local_decl.local_info = LocalInfo::StaticRef { def_id, is_thread_local };
+            match expr.kind {
+                ExprKind::StaticRef { def_id, .. } => {
+                    assert!(!this.hir.tcx().has_attr(def_id, sym::thread_local));
+                    local_decl.internal = true;
+                    local_decl.local_info = LocalInfo::StaticRef { def_id, is_thread_local: false };
+                }
+                ExprKind::Tls(def_id) => {
```

Comment:
This makes the `ThreadLocalRef` alternative even better in my mind (i.e. by comparison with `StaticRef`).